src/link/output.c

```diff
@@ -273,6 +275,47 @@ static struct SortedSection const **nextSection(struct SortedSection const **s1,
 	return (*s1)->section->org < (*s2)->section->org ? s1 : s2;
 }
 
+// Checks whether a symbol is
+static bool canStartSymName(char c)
+{
+	return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || c == '_';
+}
+
+static void printSymName(char const *name)
+{
+	for (char const *ptr = name; *ptr != '\0'; ) {
+		char c = *ptr;
+
+		if ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')
+		 || c == '_' || c == '@' || c == '#' || c == '$' || c == '.') {
```

Comment:
Extract this into `static bool canContinueSymbol(char c)`, even if this is (so far?) the only place it's called.

Reply:
I didn't factor this because it's not a strict necessity: it's possible that special characters may eventually be accepted as the first, but not anywhere else, and there's a chance we would miss the implication if only looking at `canStartSymName`.

Reply:
Alright; even without reusing `canStartSymName`, I think it would be more readable as its own function.

```c
static bool canContinueSymName(char c)
{
	return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9') ||
	       c == '_' || c == '@' || c == '#' || c == '$' || c == '.';
}
```